qiskit/quantum_info/states/statevector.py

```diff
@@ -369,6 +393,9 @@ def expectation_value(self, oper, qargs=None):
         Returns:
             complex: the expectation value.
         """
+        if isinstance(oper, Pauli):
+            return self.expectation_value_pauli(oper)
+
```

Comment:
We should add another case here for `SparsePauliOp` to loop over the Pauli terms